### PR TITLE
Fyle 4bbmph - Ionic App flow Intergration

### PR DESCRIPTION
**Pull request by iravisankar**
Added android folder
Added hooks for automate the build setup
Integration with Appflow for automate releases

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abhishek1234321) [inline android/app/src/main/AndroidManifest.xml:1]: compare the file with @tarunkumarfyle

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+unsafe-perm=true
```
Comment (abhishek1234321) [inline .npmrc:1]: @Julias0 could you check if there are any security implications

File: android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
+apply plugin: 'com.android.application'
+
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    defaultConfig {
+        applicationId "com.ionicframework.fyle595781.staging"
```
Comment (abhishek1234321) [inline android/app/build.gradle:6]: @iravisankar to take care of this

File: android/app/src/main/assets/capacitor.config.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (abhishek1234321) [inline android/app/src/main/assets/capacitor.config.json:1]: same as earlier

File: hooks/build.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+const fs = require('fs');
```
Comment (abhishek1234321) [inline hooks/build.js:1]: node version 14 and above

File: hooks/build.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+const util = require('util');
+const path = require('path');
+const stat = util.promisify(fs.stat);
+
+module.exports = function (ctx) {
+    if (!ctx.build.platform == 'android') return;
```
Comment (abhishek1234321) [inline hooks/build.js:7]: `{return;}`

File: hooks/build.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
+const fs = require('fs');
+const util = require('util');
+const path = require('path');
+const stat = util.promisify(fs.stat);
+
+module.exports = function (ctx) {
```
Comment (abhishek1234321) [inline hooks/build.js:6]: write code comments

File: hooks/build.js (hunk 6 above)
Comment (abhishek1234321) [reply hooks/build.js:7]: `if (ctx.build.platform === 'android')`

File: hooks/build.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    if (!ctx.build.platform == 'android') return;
+    const platformRoot = path.join(ctx.project.dir, 'android');
+    FILE_PATHS = {
+        "android.cameraPreview": ctx.project.dir+"/node_modules/@capacitor-community/camera-preview/android/src/main/java/com/ahm/capacitor/camera/preview/CameraPreview.java"
```
Comment (abhishek1234321) [inline hooks/build.js:10]: use template strings `${}`

File: hooks/build.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    // Commenting Manifest.permission.RECORD_AUDIO on CameraPreview.java 
+    console.log("Commenting Manifest.permission.RECORD_AUDIO on CameraPreview.java ...");
+    var cameraPreviewPath = path.resolve(process.cwd(), FILE_PATHS["android.cameraPreview"]);
+    var cameraPreviewContents = fs.readFileSync(cameraPreviewPath).toString();
+    fs.writeFileSync(cameraPreviewPath, cameraPreviewContents.replace(/Manifest.permission.RECORD_AUDIO/g, '// Manifest.permission.RECORD_AUDIO ,'), 'utf8');
+    
+
+
+    const googleCredentials = `
+    {
```
Comment (abhishek1234321) [inline hooks/build.js:22]: import this

File: hooks/prebuild.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
```
Comment (abhishek1234321) [inline hooks/prebuild.js:1]: Sentry source map stuff to be taken care of and pass the commit id in the `main.ts`

File: hooks/preserve.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+const fs = require('fs');
```
Comment (abhishek1234321) [inline hooks/preserve.js:1]: not needed

File: hooks/build.js (hunk 5 above)
Comment (abhishek1234321) [reply hooks/build.js:1]: figure out a way to build staging connected and prod connected APK

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**